### DISCUSSION TITLE
404 error when page is refreshed

### DISCUSSION THREAD

**Discussion by RahiniHugsi** [gatsbyjs/gatsby#34221] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Live website:   https://hugsi.green
Page example: https://hugsi.green/ranking/?All

Steps to replicate:
1. Hit the URL - https://hugsi.green/ranking/?All
2. Keep the browser dev tools window open
3. Refresh the page once.
4. You can see 404 error captured in browser console

Issue:
404 error is captured, though we have the targeted page active and handled.

### Reproduction Link

NA

### Steps to Reproduce

1. Hit the URL - https://hugsi.green/ranking/?All
2. Keep the browser dev tools window open
3. Refresh the page once.
4. You can see 404 error captured in browser console

### Expected Result

404 error should not be encountered for valid page URLs.

### Actual Result

404 error is captured for valid page URLs

### Environment

```shell
System:  OS:  Linux
  Binaries: 
Node: 10.24.1 x64 
npm:  6.14.12
  Languages: 
React: 16.12.0

  Browsers: 
Chrome: Version 96.0.4664.55 (Official Build) (x86_64)  
npmPackages: 
Gatsby: 2.24.62
```

**Comment by LekoArts** [comment]
Hi!

The site works fine for me so it must be something on your end/something with your specific site.
